R/apply_polygon_sample_bias.R

```diff
@@ -0,0 +1,104 @@
+#' Generate a sampling bias via a polygon
+#'
+#' This function adds a sampling bias weight column containing the sample
+#' probability based on sampling bias within a polygon.
+#'
+#' @param observations An sf object with POINT geometry.
+#'
+#' @param bias_area An sf object with POLYGON geometry. The area in which the
+#' sampling will be biased.
+#'
+#' @param bias_strength A positive numeric value. The strength of the bias to
+#' be applied in the biased area (as a multiplier). Above 1, area will be
+#' oversampled. Below 1, area will be undersampled. For example, a value of 50
+#' will result in 50 times more samples within the bias_area than outside.
+#' Conversely, a value of 0.5 will result in half less samples within the
+#' bias_area than outside.
+#'
+#' @returns An sf object with POINT geometry with a bias_weight column
+#' containing the sampling probability based on sampling bias.
+#'
+#' @export
+#'
+#' @importFrom cli cli_abort cli_warn
+#' @importFrom magrittr %>%
+#' @importFrom sf st_within
+#' @importFrom dplyr mutate
+#'
+#' @examples
+#'
+#' library(simcuber)
+#' library(sf)
+#'
+#' set.seed(123)
+#'
+#' # Create four random points
+#' n_points <- 4
+#' xlim <- c(3841000, 3842000)
+#' ylim <- c(3110000, 3112000)
+#' coordinate_uncertainty <- rgamma(n_points, shape = 5, rate = 0.1)
+#'
+#' observations_sf <- data.frame(
+#'   lat = runif(n_points, ylim[1], ylim[2]),
+#'   long = runif(n_points, xlim[1], xlim[2]),
+#'   coordinateUncertaintyInMeters = coordinate_uncertainty
+#' ) %>%
+#'   st_as_sf(coords = c("long", "lat"), crs = 3035)
+#'
+#' # Create bias_area polygon overlapping two of the points
+#' selected_observations <- st_union(observations_sf[2:3,])
+#' bias_area <- st_convex_hull(selected_observations) %>%
+#'   st_buffer(dist = 100) %>%
+#'   st_as_sf()
+#'
+#' bias_strength <- 2
+#'
+#' apply_polygon_sample_bias(observations_sf, bias_area, bias_strength)
+#'
+apply_polygon_sample_bias <- function(observations, bias_area,
+                                      bias_strength = 1) {
+  require(sf)
+
+  ### Start checks
```

Comment:
Should we have these checks here or in the `sample_observations` function?

Reply:
Good point, I'm happy to leave them in for code writing purposes until those checks are implemented in ```sample_observations``` and then they can be removed?